md/rest-api-extensions.md

```diff
@@ -1,37 +1,197 @@
+
+You can create the Index.groovy using your favorite IDE or a simple text editor. Here is the content of the file:
+```groovy
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+
+import org.bonitasoft.engine.identity.ContactData
+import org.bonitasoft.engine.identity.User
+import org.bonitasoft.engine.identity.UserCriterion
+import org.bonitasoft.web.extension.rest.RestAPIContext
+import org.bonitasoft.web.extension.rest.RestApiController
+import org.bonitasoft.web.extension.rest.RestApiResponse
+import org.bonitasoft.web.extension.rest.RestApiResponseBuilder
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import groovy.json.JsonBuilder
+
+
+
+class Index implements RestApiController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Index.class)
+
+    @Override
+    RestApiResponse doHandle(HttpServletRequest request, RestApiResponseBuilder responseBuilder, RestAPIContext context) {
+        // To retrieve query parameters use the request.getParameter(..) method.
+        // Be careful, parameter values are always returned as String values
+
+        // Retrieve p parameter
+        def p = request.getParameter "p"
+        if (p == null) {
+            return buildResponse(responseBuilder, HttpServletResponse.SC_BAD_REQUEST,"""{"error" : "the parameter p is missing"}""")
+        }
+
+        // Retrieve c parameter
+        def c = request.getParameter "c"
+        if (c == null) {
+            return buildResponse(responseBuilder, HttpServletResponse.SC_BAD_REQUEST,"""{"error" : "the parameter c is missing"}""")
+        }
+
+       // Convert parameters from string to int
+        p = p as int
+        c = c as int
+
+        // Initialize the list to store users information  
+        def usersInformation = []
+
+        // Get the list of user  
+        List<User> users = context.apiClient.identityAPI.getUsers(p*c, c, UserCriterion.FIRST_NAME_ASC)
+
+        // Iterate over each user
+        for (user in users) {
+            // Get user extra information (including email address)
+            ContactData contactData = context.apiClient.identityAPI.getUserContactData(user.id, false)
+
+            // Create a map with current user first name, last name and email address
+            def userInformation = [firstName: user.firstName, lastName: user.lastName, email: contactData.email]
+
+            // Add current user information to the global list
+            usersInformation << userInformation
+        }
+
+        // Prepare the result
+        def result = [p: p, c: c, userInformation: usersInformation]
+
+        int startIndex = p*c
+        int endIndex = p*c + users.size() - 1
+
+        // Send the result as a JSON representation
+        return buildPagedResponse(responseBuilder, new JsonBuilder(result).toPrettyString(), startIndex, endIndex, context.apiClient.identityAPI.numberOfUsers)
```

Comment:
```suggestion
        return buildPagedResponse(responseBuilder, new JsonBuilder(result).toString(), startIndex, endIndex, context.apiClient.identityAPI.numberOfUsers)
```

Reply:
ℹ️ This is to reduce the response size and the same should be applied for the other example (see line 248)